drivers/SmartThings/matter-switch/src/init.lua

```diff
@@ -840,14 +841,26 @@ end
 local function hue_attr_handler(driver, device, ib, response)
   if ib.data.value ~= nil then
     local hue = math.floor((ib.data.value / 0xFE * 100) + 0.5)
-    device:emit_event_for_endpoint(ib.endpoint_id, capabilities.colorControl.hue(hue))
+    local current_color_mode = get_field_for_endpoint(device, COLOR_MODE, ib.endpoint_id)
+    -- don't send capability events if the color of the device is being determined by CurrentX and CurrentY (1) or ColorTemperatureMireds (2)
+    if current_color_mode == 1 or current_color_mode == 2 then
```

Comment:
This might be more clear just changing this to
```suggestion
    if current_color_mode != 0 then
```

Reply:
I did it this way in case a device doesn't implement the ColorMode attribute. In that case, `COLOR_MODE` will be nil in the driver, so doing the logic this way maintains the same behavior (we only want to prevent capabilities from being emitted if ColorMode is actually implemented)